app/models/spree/order_decorator.rb

```diff
@@ -191,5 +191,54 @@ def flow_payment_method
         'cc' # creait card is default
       end
     end
+
+    def flow_customer_email
+      flow_data['order']&.[]('customer')&.[]('email')
+    end
+
+    def flow_ship_address
+      flow_destination = flow_data['order']&.[]('destination')
+      return unless flow_destination.present?
+
+      flow_contact = flow_destination['contact']
+      s_address = ship_address || build_ship_address
+      s_address.attributes = { first_name: flow_contact['name']['first'], last_name: flow_contact['name']['last'],
```

Comment:
This hash creation and line 218 hash creation are looking very similar - could be extracted into a parametrized method.

Reply:
Agreed!!